kythe/cxx/indexer/textproto/analyzer.cc

```diff
@@ -363,6 +365,83 @@ absl::Status TextprotoAnalyzer::AnalyzeAny(
   return AnalyzeMessage(file_vname, *value_proto, *msg_desc, parse_tree);
 }
 
+// Trims whitespace (including newlines) and comments from the start of the
+// input.
+void ConsumeTextprotoWhitespace(re2::StringPiece* sp) {
+  re2::RE2::Consume(sp, R"((\s+|#[^\n]*)*)");
```

Comment:
IMO, using a regexp is a bit overkill here rather than something like `absl::StripLeadingAsciiWhitespace`, but 🤷‍♂️

Reply:
Is there an easy way to also parse comments with absl/strings?

Reply:
Not exactly, but line-comments are pretty straight-foward:
```
input = absl::StripLeadingAsciiWhitespace(input);
if (absl::ConsumePrefix(&input, "#")) {
  if (auto end = input.find("\n"); end != input.npos) {
    input.remove_prefix(end + 1);
  } else {
    input.remove_prefix(input.size());
  }
}
return input;
```

The regex is likely fine, though, (hence the earlier shrug).

Reply:
Sounds good, thanks.